connector-sdk/core/src/main/java/io/camunda/connector/api/secret/SecretProvider.java

```diff
@@ -30,4 +32,9 @@ public interface SecretProvider {
    *     returned.
    */
   String getSecret(String name);
+
+  /**
+   * @return
+   */
+  List<String> getSecretValues();
```

Comment:
As discussed we dont need to change the `SecretProvider` interface. We can extract all used secrets from the `ActivatedJobs` json and only fetch the secrets that were used.